src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Synthesized.cs

```diff
@@ -652,6 +652,255 @@ internal class C2<[Attr] T2> { }
             Assert.Equal(new[] { "Attr" }, GetAttributeNames(typeParam.GetAttributes()));
         }
 
+        [Fact]
+        [WorkItem(46439, "https://github.com/dotnet/roslyn/issues/46439")]
+        public void RecordSynthesizedMembers()
```

Comment:
nit: Consider also validating that methods provided explicitly by the user don't get the compiler-generated attribute.